pystiche_papers/sanakoyeu_et_al_2018/_data.py

```diff
@@ -0,0 +1,129 @@
+from typing import Any, Dict, Optional, Sized, cast
+
+import torch
+from torch.utils.data import DataLoader, Dataset, Sampler
+
+from pystiche.data import ImageFolderDataset
+from pystiche.image.transforms import ComposedTransform, Transform, ValidRandomCrop
+from pystiche.image.transforms import functional as F
+from pystiche.image.utils import extract_image_size
+
+from ..data.utils import FiniteCycleBatchSampler
+from ..utils import OptionalGrayscaleToFakegrayscale
+
+
+def image_transform(edge_size: int = 768,) -> ComposedTransform:
+    class OptionalRescale(Transform):
```

Comment:
Since this has multiple branches, I suggest we move it outside of `image_transform` to make the testing easier.